### PR TITLE
Player: re-add busy dialog on start

### PR DESCRIPTION
**Pull request by FernetMenta**
- brings back busy dialog
- add new event for signalling start of audio and video streams
- fix potential error pop-up in case of videoplayer was stopped.
- move python callback OnPlayback.. to main thread

### CODE REVIEW

File: xbmc/cores/RetroPlayer/RetroPlayer.cpp [hunk 1]
```diff
@@ -187,6 +187,7 @@ bool CRetroPlayer::OpenFile(const CFileItem& file, const CPlayerOptions& options
     RegisterWindowCallbacks();
     SetSpeedInternal(1.0);
     m_callback.OnPlayBackStarted(fileCopy);
+    m_callback.OnAVStarted(fileCopy);
```
Comment (FernetMenta) [inline xbmc/cores/RetroPlayer/RetroPlayer.cpp]: @garbear could you please have a look. You may want to move OnAVStarted to another pos when av streams are ready. CApp shows a busy dialog the time between OnPlayBackStarted and OnAVStarted. Not sure if this is relevant for RP.

Comment (garbear) [reply xbmc/cores/RetroPlayer/RetroPlayer.cpp]: In RetroPlayer, A/V streams are independent. Should this callback be called for audio start, video start, or both?

Comment (FernetMenta) [reply xbmc/cores/RetroPlayer/RetroPlayer.cpp]: it should be called only once. VP fires the event after audio and video streams are ready to go and stream players are unpaused. CApp closes the busy dialog on this event.

Comment (garbear) [reply xbmc/cores/RetroPlayer/RetroPlayer.cpp]: We could defer this call until the first frame is shown, at the expense of additional complexity. IMO it's not worth it, so placing the callback here is fine.

Comment (garbear) [reply xbmc/cores/RetroPlayer/RetroPlayer.cpp]: Let's avoid adding player logic to the AV streams. Invoking the callback here is fine.

**Review by garbear — APPROVED** [review]

**Comment by FernetMenta** [comment]
build errors unrelated

**Comment by iz8mbw** [comment]
Hi @FernetMenta 
is this PR related to https://trac.kodi.tv/ticket/17819 ?

**Pull request merged**